apps/roam/src/components/DiscourseNodeSearchMenu.tsx

```
@@ -0,0 +1,560 @@
+      );
+
+      setDiscourseTypes(allNodeTypes);
+
+      const initialCheckedTypes: Record<string, boolean> = {};
+      allNodeTypes.forEach((t) => {
+        initialCheckedTypes[t.type] = true;
+      });
+      setCheckedTypes(initialCheckedTypes);
+
+      const initialSearchResults = allNodeTypes.reduce(
+        (acc, type) => ({ ...acc, [type.type]: [] }),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)